scripts/generate_secrets.py

```diff
@@ -11,23 +19,75 @@ def get_file_contents(folder, filename, trim=False):
+    pub_bytes = pubilc_key.public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+
+    kid = generate_kid_from_key(pub_bytes)
+    dict[purpose][kid] = key
+
+
+def add_private_key_to_dict(dict, purpose, key):
+
+    private_key = load_pem_private_key(key.encode(), password="digitaleq".encode(), backend=backend)
+
+    pub_key = private_key.public_key()
+
+    pub_bytes = pub_key.public_bytes(Encoding.PEM, PublicFormat.SubjectPublicKeyInfo)
+
+    kid = generate_kid_from_key(pub_bytes)
+    dict[purpose][kid] = key
+
+
 if len(sys.argv) < 2:
     raise Exception("Please specify the directory containing the secrets")
 
 keys_folder = sys.argv[1]
 
-keys = {}
+public_keys = {
+    PURPOSE_AUTHENTICATION: {},
+    PURPOSE_SUBMISSION: {},
+}
 
-keys['EQ_SUBMISSION_SDX_PUBLIC_KEY'] = get_file_contents(keys_folder, 'sdc-submission-encryption-sdx-public-key.pem')
-keys['EQ_SUBMISSION_SR_PRIVATE_SIGNING_KEY'] = get_file_contents(keys_folder, 'sdc-submission-signing-sr-private-key.pem')
-keys['EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY'] = get_file_contents(keys_folder, 'sdc-user-authentication-signing-rrm-public-key.pem')
-keys['EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY'] = get_file_contents(keys_folder, 'sdc-user-authentication-encryption-sr-private-key.pem')
+add_public_key_to_dict(public_keys, PURPOSE_SUBMISSION, get_file_contents(keys_folder, 'sdc-submission-encryption-sdx-public-key.pem'))
+add_public_key_to_dict(public_keys, PURPOSE_AUTHENTICATION, get_file_contents(keys_folder, 'sdc-user-authentication-signing-rrm-public-key.pem'))
 
-if os.path.exists(os.path.join(keys_folder, 'sdc-user-authentication-encryption-sr-public-key.pem')):
-    keys['EQ_USER_AUTHENTICATION_SR_PUBLIC_KEY'] = get_file_contents(keys_folder, 'sdc-user-authentication-encryption-sr-public-key.pem')
+# Support depricated kid value until it is no longer used
+public_keys[PURPOSE_AUTHENTICATION]['EDCRRM'] = get_file_contents(keys_folder, 'sdc-user-authentication-signing-rrm-public-key.pem')
 
-if os.path.exists(os.path.join(keys_folder, 'sdc-user-authentication-signing-rrm-private-key.pem')):
-    keys['EQ_USER_AUTHENTICATION_RRM_PRIVATE_KEY'] = get_file_contents(keys_folder, 'sdc-user-authentication-signing-rrm-private-key.pem')
+private_keys = {
+    PURPOSE_AUTHENTICATION: {},
+    PURPOSE_SUBMISSION: {},
+}
+
+add_private_key_to_dict(private_keys, PURPOSE_AUTHENTICATION, get_file_contents(keys_folder, 'sdc-user-authentication-encryption-sr-private-key.pem'))
+add_private_key_to_dict(private_keys, PURPOSE_SUBMISSION, get_file_contents(keys_folder, 'sdc-submission-signing-sr-private-key.pem'))
+
+# Support default value until all upstream systems are passing kid
+private_keys[PURPOSE_AUTHENTICATION]['EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY'] = \
+    get_file_contents(keys_folder, 'sdc-user-authentication-encryption-sr-private-key.pem')
+
+keys = {
+    "public_keys": public_keys,
+    "private_keys": private_keys,
+}
 
 with open('secrets.yml', 'w') as f:
```

Comment:
This file is getting more complex, should we document the structure of it somewhere, perhaps in the new design/architecture decisions log or README?